cg/cli/upload/delivery_report.py

```diff
@@ -20,14 +21,14 @@
 @click.pass_context
 def upload_delivery_report_to_scout(context: click.Context, case_id: str, dry_run: bool):
     """Fetches a delivery report from housekeeper and uploads it to scout."""
-
     click.echo(click.style("--------------- DELIVERY REPORT UPLOAD ---------------"))
-
     case: Family = get_report_case(context, case_id)
     report_api: ReportAPI = get_report_api(context, case)
-    report_path: str = report_api.get_delivery_report_from_hk(case_id)
-
-    if not dry_run:
+    version: Version = report_api.housekeeper_api.last_version(case_id)
+    delivery_report: Optional[str] = report_api.get_delivery_report_from_hk(
+        case_id=case_id, version=version
+    )
+    if delivery_report and not dry_run:
```

Comment:
```suggestion
    if not dry_run:
```
If think it's better to let if fail if it doesn't find a delivery report OR to explicitly add an else statement to log that the report was not uploaded to scout. Otherwise we risk missing uploads silently, don't we?

Reply:
This command is to upload manually the delivery reports to Scout, it's independent of the automation, so we won't miss silently the upload in any case. However, I added an additional log ([9e06535](https://github.com/Clinical-Genomics/cg/pull/2256/commits/9e065359df4b7ed88c0797ff433a2fe60598ce58)).

I've also added a re-upload option so we have more control over the Scout commands. It was forcing the re-upload by default ([9a04fe2](https://github.com/Clinical-Genomics/cg/pull/2256/commits/9a04fe221b1855ca259b78b6fa2f204afbf316a0)).